### DISCUSSION TITLE
Engine force gets reset to 0 at the start of the round, but the reactor's output is carried over, causing overvoltage

### DISCUSSION THREAD

**Discussion by NotWendy** [FakeFishGames/Barotrauma#15125] [Bug reports]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [ ] My issue happened while using mods.

### What happened?

In the campaign, if your engine is drawing a lot of power from the grid at the moment when you successfully finish the round, you're at a risk of immediately burning to death when the next round begins.
This happens because the game resets the engine's force (and its power consumption) to 0 at the start of the next round, but does not automatically adjust reactor's output the same way.
So when you spawn in, your reactor will be producing a lot more power than necessary, and its automatic control often doesn't have enough time to handle the situation before the junction boxes break from overvoltage.

I'm marking this as a multiplayer bug because I don't play singleplayer enough to know if it handles this situation any differently.

### Reproduction steps

The easiest way to experience this problem is to pass at full speed through a natural formation at the end of the level. Without having to dock, you can comfortably maintain full speed as you open the map and switch to the next level.

1. Beat a level in the campaign which ends with a natural formation and maintain full speed as you switch levels.
2. At the start of the round your submarine will be stationary, maintaining position with 0 engine force.
3. Observe the immediate overvoltage as the reactor tries to adjust to the sudden decrease in load. The more power your engine draws from the grid at full force and the lower your overvoltage resistance is, the more noticeable the effect is. Ranging from minor overvoltage to immediate junction box fires across the submarine.

### Bug prevalence

Happens regularly

### Single player or multiplayer?

Multiplayer hosted from the in-game menu (= using a listen server)

### Version

v1.6.19.1 (Unto the Breach Update Hotfix 2)

### Which operating system did you encounter this bug on?

Windows

**Comment by Regalis11** [comment]
Addressed in https://github.com/FakeFishGames/Barotrauma-development/commit/3db0da2454514bd3748b955f1f54056c63be774a (commit in our private repo)

> Measures to prevent overvoltage from damaging the sub at the start of a round. The load on the electrical grid often changes between rounds (due to e.g. engines resetting to 0, and the sub receiving power through an outpost's docking port), and the reactor may not react to the changes fast enough. Now the reactor does rapid automatic adjustments in the first 5 seconds of the round (unless it's being operated by a player), and overvoltage doesn't damage devices during those 5 seconds.

**Reply by NotWendy** [reply]
Sounds good, that should cover these situations nicely.

**Reply by NotWendy** [reply]
Actually, @Regalis11, could this theoretically be used to easily fry an outpost's grid in those first 5 seconds, if the reactor is in manual mode? If your junction boxes are invulnerable to overvoltage, but the outpost's aren't, you could pump tons of excess power into the grid.
Overvolting outposts should probably be treated as a separate issue (potential issue?), it's just something that came to my mind.
Or has outpost overvoltage already been addressed at some point and I'm just not up to date with the development news?

**Reply by Mixmaster49** [reply]
@NotWendy I just tested this with a garbage sub I made and it definitely works. It even started a fire on the outpost.

**Reply by NotWendy** [reply]
So that's probably not a good thing. But it'd still be a separate ticket, I guess.

**Reply by Regalis11** [reply]
Good points! I would consider that a separate issue though, and imo the 5 second duration seems so short that in practice it'd be quite difficult to do much damage within that time (considering you need to get to the reactor first too).

**Reply by NotWendy** [reply]
Maybe the automatic adjustments could be bypassed with wiring components. It's a lot of work just to cheese an outpost, but it's not like anyone ever raids those in a legit way, by getting into a shootout with the inhumanly durable guards.
Anyways, yea, cheesing outposts is a separate issue. Or rather an entire set of them. Using your submarine's reactor to fry the grid is just one of numerous things people do to get to that free, infinitely respawning loot.

**Comment by Regalis11** [comment]
This has been addressed in our private repo now, fix coming up in the next unstable build.
